great_expectations/data_context/store/html_site_store.py

```diff
@@ -386,6 +389,31 @@ def copy_static_assets(self, static_assets_source_dir=None):
                 __file__, os.path.join("..", "..", "render", "view", "static")
             )
 
+        # If `static_assets_source_absdir` contains the string ".zip/" (unix) or ".zip\" (win)
+        # then we check if Great Expectations has been installed inside a zip file (see PEP 273)
+        # If so, we need to extract all the static files and run this function again
+        static_assets_source_absdir = os.path.abspath(static_assets_source_dir)
+        zip_re = re.match(
+            f"(.+[.]zip){re.escape(os.sep)}(.+)",
+            static_assets_source_absdir,
+            flags=re.IGNORECASE,
+        )
+
+        if zip_re:
+            zip_filename = zip_re.groups()[0]  # e.g.: /home/joe/libs/my_python_libs.zip
+            path_in_zip = zip_re.groups()[1]  # great_expectations/render/view/static
+            unzip_destination = tempfile.mkdtemp()
+
+            if is_zipfile(zip_filename):
+                with ZipFile(zip_filename) as zipfile:
+                    static_files_to_extract = [
+                        file
+                        for file in zipfile.namelist()
+                        if file.startswith(path_in_zip)
+                    ]
+                    zipfile.extractall(unzip_destination, static_files_to_extract)
+                return self.copy_static_assets(unzip_destination)
+
```

Comment:
Since this logic is very specific to zip-files, could we extract the logic into a separate method? Maybe something like this:

```python
def _handle_zipped_static_assets(self) -> bool:
       # If `static_assets_source_absdir` contains the string ".zip/" (unix) or ".zip\" (win)
        # then we check if Great Expectations has been installed inside a zip file (see PEP 273)
        # If so, we need to extract all the static files and run this function again
        static_assets_source_absdir = os.path.abspath(static_assets_source_dir)
        zip_re = re.match(
            f"(.+[.]zip){re.escape(os.sep)}(.+)",
            static_assets_source_absdir,
            flags=re.IGNORECASE,
        )

        if zip_re:
            zip_filename = zip_re.groups()[0]  # e.g.: /home/joe/libs/my_python_libs.zip
            path_in_zip = zip_re.groups()[1]  # great_expectations/render/view/static
            unzip_destination = tempfile.mkdtemp()

            if is_zipfile(zip_filename):
                with ZipFile(zip_filename) as zipfile:
                    static_files_to_extract = [
                        file
                        for file in zipfile.namelist()
                        if file.startswith(path_in_zip)
                    ]
                    zipfile.extractall(unzip_destination, static_files_to_extract)
                self.copy_static_assets(unzip_destination)
                return True
       return False
```

That way, we can isolate the zip-specific code and simply invoke it in `copy_static_assets`
```python
unzipped = self._handle_zipped_static_files()
if unzipped:
    return
```

Thoughts?